dev/com.ibm.ws.org.apache.cxf.cxf.rt.rs.client.3.2/src/org/apache/cxf/jaxrs/client/spec/InvocationBuilderImpl.java

```diff
@@ -257,6 +257,7 @@ public Builder header(String name, Object value) {
 
     @Override
     public Builder headers(MultivaluedMap<String, Object> headers) {
+        webClient.removeAllHeaders();
         RuntimeDelegate rd = HttpUtils.getOtherRuntimeDelegate();
         for (Map.Entry<String, List<Object>> entry : headers.entrySet()) {
```

Comment:
The API for Invocation.Builder#headers() states that passing a null value in for headers needs to remove all existing headers:
https://jax-rs.github.io/apidocs/2.1/javax/ws/rs/client/Invocation.Builder.html#headers-javax.ws.rs.core.MultivaluedMap-

At line 262, passing in null will cause an NPE when headers.entrySet() is invoked so a null check will also be necessary.